src/components/DiagnosticsBody.vue

```diff
@@ -62,14 +97,32 @@ export default Vue.extend({
     <sortable-table
       key-field="id"
       :headers="headers"
-      :rows="rows"
+      :rows="filteredRows"
       :search="false"
       :table-actions="false"
       :row-actions="false"
       :sub-rows="true"
       :sub-expandable="true"
       :sub-expand-column="true"
     >
+      <template #no-rows>
+        <td :colspan="headers.length + 1">
+          <empty-state
+            icon="icon-search"
+            heading="No results found"
+            body="Try showing muted diagnostics to see some results."
+          >
+            <template #primary-action>
+              <button
+                class="btn role-primary"
+                @click="toggleMute"
+              >
+                Show Muted
```

Comment:
Clicking on this button doesn't update the state of the "Hide Muted" button at line 90 (and I still don't know how you modern UI experts can identify widgets without ID attributes).

Reply:
This is a `ToggleSwitch` bug in `@rancher/components`. Created rancher/dashboard#6741 to address. 

The gist of the problem is that `ToggleSwitch` is written in such a way that it uses the `value` prop as an initial state to manage state internally. This means that any updates to the `value` prop will not be reflected in the rendered output..

Reply:
Could you wrap the variable with a setter/getter and do something in the setter?

Reply:
Not quite, `ToggleSwitch` manages its own state after using the `value` prop as an initial value. 

What's really required is that `ToggleSwitch` reacts to changes coming in through the `value` prop. This is how we would expect components to behave anyway, unless otherwise noted fro special cases. 

Raised rancher/dashboard#6751 to resolve the issue and will publish an update to `@rancher/components` after merging.